### DISCUSSION TITLE
Trigger script on button press

### DISCUSSION THREAD

**Discussion by legitleavitt** [gopro/labs#702] [Q&A]
Hey all, I have a blockly script I'd like my gopro to iterate through upon some signal, but I can't use a lot of the standard triggers (motion, USB power, time, etc.). I want to know if there's a way to call a saved script when an action is taken, ideally from the Quik app?

**An example scenario:**
The camera is on my roof not recording, and after confirming that I like the angle via the quik app live view, I hit record/switch modes/disconnect/something, the camera STILL isn't recording, and then it loads startRecording script, changes some settings, sleeps for 1 minute, and then wakes up and starts recording.

**What I've tried**
I was able to use an if/else loop to check to see if the camera is recording, and then sleeps for 9 seconds and loops to check again. But if the camera's already recording, I can't change the settings I want to change.
I tried the above loop with "App (not)connected" which 'does' work, but in order to disconnect I have to turn off bluetooth AND wifi, or quit the quik app, not ideal.

Appreciate any ideas y'all might have!

**Comment by dnewman-gpsw** [comment]
Quik and Labs have no connection, no easy way around that.  Open to ideas, but any changes to Quik is out of scope for Labs.

**Reply by legitleavitt** [reply]
I was thinking if there'd be a way to trigger a script via a signal that Quik sends by default. Like if labs could co-opt a record button press to execute a script instead of starting recording, or treat a mode switch command from the app as pushing the mode button on the side.
Actually, triggering off a mode switch with the condition of app connected could be a viable route. Have a loop that checks the current mode, and if it switches to photo, change it to video and hit record perhapse?

**Reply by dnewman-gpsw** [reply]
Currently there is no test for current mode, but that would be the most straightforward approach.  If(mode_photo) then ...
